src/Notifier.js

```diff
@@ -133,29 +153,35 @@ var Notifier = {
         }
 
         if(enable) {
-            if (!this.havePermission()) {
-                global.Notification.requestPermission(function() {
-                    if (callback) {
-                        callback();
+            // Case when we do not have the permission as 'granted'
+            if (this.isPermissionDefault()) {
+                // Attempt to get permission from user
+                var self = this;
+                global.Notification.requestPermission().then(function(result) {
```

Comment:
So I shall shift this to callback one? I think that MDN doc is little updated, as this is supported in chrome now.